### ISSUE TITLE
Code explanation

### ISSUE THREAD

**Issue by luisegehaijing** [dimonenka/RL_policy_budget#1]
Hi, there. Your work is appealing to me and I am trying to understand your code. 
Could you kindly explain how you calculate the social welfare?
In particular,  why is the self.SW updated like this?

self.last_rewards_buffer += rewards
                if any(dones):
                    for i, d in enumerate(dones):
                        if d:
                            self.last_rewards[i] = self.last_rewards_buffer[i].copy()
                            self.last_rewards_buffer[i] *= 0
                  
                    SW = (self.last_rewards * assignment_mask.detach().numpy()).sum().item()
                    
                    self.SW = 0.99 * self.SW + 0.01 * SW  

Thank you very much.

**Comment by dimonenka** [comment]
Hi Luise,

self.SW is just used for logging (in the .log() method). It is estimated as a moving average of returns in the episode; the return itself is a sum of rewards in the episode over all agents w.r.t. their assignments to policies (eg in the EM algorithm, each agent is fully assigned to one policy).

Let me know if any other questions remain, I'll be happy to help.

**Comment by luisegehaijing** [comment]
I see. Could you explain where the weight 0.99 come from? Many thanks!

**Comment by dimonenka** [comment]
The weight doesn't really matter, it just smoothes the social welfare metric when logging.
